dev-docs/source/WritingPerformanceTests.rst

```diff
@@ -134,6 +134,37 @@ Best Practice Advice
 -  Always perform test set-up outside of the test method. That way your
    timings will only relate to the target code you wish to measure.
 
+Avoiding Compiler Optimizations
+###############################
+
+A common issue with performance tests is when you want to compute some
+value over a large number of iterations but ultimately you don't want
+to actually use it in the end. We have included some handy utility functions
+to avoid the compiler optimizing unused variables away. For more information
+this conference talk is quite enlightening: https://youtu.be/nXaxk27zwlk?t=2441.
+
+For example, we attempt to time the sum of a value after many calls to an operation:
+
+.. code-block::
```

Comment:
This code block errored for me, so didn't appear in the built docs

`WARNING: Error in "code-block" directive:
1 argument(s) required, 0 supplied.`

Reply:
Thanks. This should be fixed now.